### PR TITLE
feat: Add prefer option to prefer-class-directive

### PR DESCRIPTION
**Pull request by sdarnell**
This the prefer option can be 'empty' or 'always' with 'empty' being the default. 'always' matches the current behaviour, always insisting on directives 'empty' only prefers directives when one class is empty.

### CODE REVIEW

**Comment by sdarnell** [comment]
Fixes #689

File: docs/rules/prefer-class-directive.md [hunk 1]
```diff
@@ -22,13 +22,16 @@ This rule aims to replace a class with ternary operator with the class directive
 
 ```svelte
 <script>
-  /* eslint svelte/prefer-class-directive: "error" */
+  /* eslint svelte/prefer-class-directive: "empty" */
```
Comment (ota-meshi) [inline docs/rules/prefer-class-directive.md:25]: I think the configuration is probably wrong.

```suggestion
  /* eslint svelte/prefer-class-directive: ["error", {"prefer": "empty"}] */
```

File: tests/fixtures/rules/prefer-class-directive/invalid/_config.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+	"options": [{ "prefer": "always" }]
+}
```
Comment (ota-meshi) [inline tests/fixtures/rules/prefer-class-directive/invalid/_config.json:3]: Could you please add test cases to check what is reported as an error when using `empty`?

**Review by ota-meshi — CHANGES_REQUESTED** [review]
Thank you for this PR!

File: docs/rules/prefer-class-directive.md (hunk 1 above)
Comment (sdarnell) [reply docs/rules/prefer-class-directive.md]: Fixed. Thanks.

File: tests/fixtures/rules/prefer-class-directive/invalid/_config.json (hunk 2 above)
Comment (sdarnell) [reply tests/fixtures/rules/prefer-class-directive/invalid/_config.json:3]: I've added some specific tests, expanding on test 01.
But note that as `prefer: "empty"` is a relaxation of the rule, the other tests cover it quite well.
While doing that I noticed that the 'empty' rule should also apply when the classname contains just whitespace - so I rejigged my changes to handle that too.

**Review by ota-meshi — APPROVED** [review]
Thank you! LGTM!
